### DISCUSSION TITLE
GitHub API "combined status" is always Pending

### DISCUSSION THREAD

**Discussion by supercobra** [community/community#58407] [Apps, API and Webhooks]
### Select Topic Area

Bug

### Body

I am trying to get the status of a repo. From Github doc it should be one of the following:

> additionally, a combined state is returned. The state is one of:
> 
> failure: if any of the contexts report as error or failure 
> pending: if there are no statuses or a context is pending 
> success: if the latest status for all contexts is success

This is very useful for checking if an automated build passed or failed.

Using this API to get a repo combined status:

https://docs.github.com/en/rest/commits/statuses?apiVersion=2022-11-28#get-the-combined-status-for-a-specific-reference

```
curl -L \
  -H "Accept: application/vnd.github+json" \
  -H "Authorization: Bearer $KEY"\
  -H "X-GitHub-Api-Version: 2022-11-28" \
  https://api.github.com/repos/supercobra/st/commits/main/status
Response:
```

```
{
  "state": "pending",
  "statuses": [

  ],
  "sha": "dbf9e48d06917c06b859823008444f3baebd58f4",
  "total_count": 0,
...
```
Github response is always a pending state regardless of the repo so I am inclined to think that there is a bug in the Github API. Anybody has a fix or workaround?

**Comment by CodyGirl** [comment]
Is there update with this bug or is there another way to get the state of a branch?

**Reply by supercobra** [reply]
Stackoverflow issue: https://stackoverflow.com/questions/76502504/github-api-combined-status-is-always-pending

**Comment by Lms24** [comment]
Is this supposed to be fixed or is there an alternative (working) endpoint we can query to get the combined status of a commit?

**Comment by CodyGirl** [comment]
I tried graphQL API as well but that just returns me null.
query {
    repository(owner:"<owner_name>", name:"<repo_name>") {
      pullRequest(number:1){
        commits(last:1){
          nodes{
            commit{
              status{
                state
              }
            }
          }
        }
      }
    }
}

**Comment by MerzDaniel** [comment]
I reported this as a bug using the GithubSupport (https://support.github.com/contact/bug-report). Let's hope they fix it soon

**Reply by CodyGirl** [reply]
Hi @MerzDaniel , is there any progress on the bug ticket?

**Comment by wechuli** [comment]
> Github response is always a pending state regardless of the repo so I am inclined to think that there is a bug in the Github API. Anybody has a fix or workaround?

For the particular example quoted, I don't see that there are any commit statuses on that commit - https://github.com/supercobra/st/commit/dbf9e48d06917c06b859823008444f3baebd58f4. Unless am mistaken, it is the documented behavior that the API would return a `pending` status when no commit statuses exist on the commit:

> pending: if there are no statuses or a context is pending

For the others running into the same issue, you might want to confirm whether whatever integration you are using is reporting Checks rather than commit Statuses. For example, if you are using GitHub Actions for your CI/CD, it will report Checks not commit Statuses. You can use the [List check runs for a Git reference](https://docs.github.com/en/rest/checks/runs?apiVersion=2022-11-28#list-check-runs-for-a-git-reference) endpoint to view Checks on a commit sha. If both of these endpoints don't return any values and you can see that the commit status or check was reported on the commit in the UI, then definitely contact GitHub Support.
